core/src/avm2/regexp.rs

```diff
@@ -204,17 +209,68 @@ impl<'gc> RegExp<'gc> {
         ret
     }
 
-    pub fn replace(
+    // Implements string.replace(regex, replacement) where the replacement is
+    // a function.
+    pub fn replace_fn(
+        &mut self,
+        activation: &mut Activation<'_, 'gc, '_>,
+        text: AvmString<'gc>,
+        f: &FunctionObject<'gc>,
+    ) -> AvmString<'gc> {
+        self.replace_with_fn(activation, &text, |activation, txt, m| {
+            let mut args = std::iter::once(Some(&m.range))
+                .chain((m.captures.iter()).map(|x| x.as_ref()))
+                .map(|o| match o {
+                    Some(r) => Value::from(AvmString::new(
+                        activation.context.gc_context,
+                        &txt[r.start..r.end],
+                    )),
+                    None => Value::from(AvmString::default()),
+                })
+                .collect::<Vec<_>>();
+            args.push(Value::from(m.range.start));
```

Comment:
Is there any reason for using `.chain()` before and `.push()` here? I think it could be done with either method, but mixing them up feels a bit odd to me. Probably the `.chain()` method is slightly more efficient due to its ability to pre-allocate `Vec` storage, while `.push()` may be more readable, at least in my opinion. Anyway I don't have a strong preference to either side, just prefer to stick with one.